_posts/en/newsletters/2020-04-29-newsletter.md

```diff
@@ -0,0 +1,265 @@
+
+## Action items
+
+- **Review the disclosure of a potential LN issue:** as explained in the
+  *news* section below, a public disclosure this week describes a new
+  method for stealing money from LN routing nodes (unaffected are
+  endpoint nodes that are only the initial spenders or final receivers
+  of payments).  The issue partly overlaps with a previous well-known
+  issue that has not been exploited (to our knowledge) because almost
+  all onchain transactions relayed in the past two years confirmed
+  relatively quickly even if they only paid the default minimum relay
+  feerate.  If feerates increase significantly for an extended period of
+  time, these issues will become more critical.  See our explanation in
+  the *news* section for details and contact your LN software vendor if
+  you have concerns about how the this issue might affect your channels.
+
+## News
+
+<!-- break atomicity == fission attack? -->
+
+- **New attack against LN payment atomicity:** Matt Corallo started a
+  thread across both the [Lightning-Dev][corallo thread ld] and
+  [Bitcoin-Dev][corallo thread bd] mailing lists disclosing an attack
+  discovered during [discussion][BOLTs #688] about allowing LN
+  commitment transactions to be [CPFP][topic cpfp] fee bumped via
+  [anchor outputs][topic anchor outputs].  We'll describe the attack
+  using an extended example: Alice uses an LN channel to send Bob a Hash
+  Time Lock Contract (HTLC) which is designed to be settled in either of
+  the following ways:
+
+     - If Bob discloses the preimage for `<hash>`, he can spend 1 BTC of
+       Alice's money
+
+     - Otherwise, after 80 blocks, Alice can spend that 1 BTC back to
+       herself
+
+    Alice also told Bob that the goal of her payment is to pay Mallory, so Bob
+    uses a channel he has with Mallory to send her a related HTLC:
+
+     - If Mallory discloses the preimage for `<hash>`, she can spend 1
+       BTC of Bob's money (we're ignoring routing fees in this example)
+
+     - Otherwise, after 40 blocks, Bob can spend his 1 BTC back to
+       himself
+
+    Although the above HTLCs are usually created and settled offchain,
+    each party also has a *commitment transaction* they can use to
+    put the HTLC commitment onchain.  A separate onchain *settlement
+    transaction* can fulfill either condition of the HTLC.
+
+    For example, Mallory can publish the commitment transaction and then
+    create a settlement transaction that provides the preimage and
+    claims Bob's 1 BTC.  If Bob sees Mallory's preimage settlement
+    transaction before the 80 block timeout from the Alice-Bob contract,
+    Bob can extract that preimage and use it to claim the 1 BTC from
+    Alice (either onchain or offchain).  Or, if Bob doesn't see a
+    preimage settlement transaction, Bob can create his own refund
+    settlement transaction after 40 blocks that takes back his 1 BTC,
+    allowing him to also refund Alice her 1 BTC (again, either onchain
```

Comment:
Dunno if refund exactly matches here, it's more "allowing him safely to cancel back Alice's offer of 1 BTC. Either cooperatively offchain (`update_fail_htlc`) or at Alice's initiative onchain (`HTLC-timeout`). Refund let think Bob had custody of Alice coin, which he doesn't without preimage knowledge.

Reply:
I agree the phrase "refund" as typically used in commerce doesn't exactly match here.  Perhaps the phrasing most similar to existing commerce terminology would be a "release" from a credit/debit card hold-and-release process.  However, I think "refund" is the term that's been used for this concept during the historic development of payment channels:

- "initial funding transaction and time-locked refund"  --Jeremy Spillman, "Spillman channels", https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2013-April/002433.html

- "Non-interactive time-locked refunds"  --Peter Todd, BIP65 CLTV payment channels, https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki#non-interactive-time-locked-refunds

- "In effect, one desires to construct a payment which is contingent upon knowledge of R by the recipient within a certain  timeframe.  After this timeframe, the funds are refunded back to the sender." --Thaddeus Dryja & Joseph Poon, The Bitcoin Lightning Network:Scalable Off-Chain Instant Payments (version 0.5.9.2), https://lightning.network/lightning-network-paper.pdf

Reply:
Okay thanks for pointers, that makes sense. In this case I would turn sentence differently, "allowing him to signal to Alice she can refund herself"?. Non-cooperative case scenario you don't need Bob to be rightfully refunded.